CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(sail_riscv)
+
+# Enable CTest
+enable_testing()
+
+# We technically require C++20 since the C generated by Sail - which we compile
+# as C++ - uses designated initialisers, a C++20 feature. However in practice
+# much older compilers support this feature so everything does work with C++17,
+# but you get lots of warnings on compilers that support C++20 if you use
+# this feature without -std=c++20.
+if (cxx_std_20 IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+    set(CMAKE_CXX_STANDARD 20)
+else()
+    set(CMAKE_CXX_STANDARD 17)
+endif()
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
+# Export compile_commands.json for IDE support.
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+
+# Always use Position Independent Code. By default it is only used for
+# shared libraries (which require it), but you also need it for static
+# libraries if you link them into shared libraries.
+# Generally it just simplifies everything for a negligable performance cost.
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
+# Don't allow undefined symbols in shared libraries. This is generally a pain.
+if (UNIX)
```

Comment:
I think it would be cleaner to use https://cmake.org/cmake/help/latest/module/CheckLinkerFlag.html

```
check_linker_flag(C -Wl,--no-undefined LINKER_SUPPORTS_NO_UNDEFINED)
if (LINKER_SUPPORTS_NO_UNDEFINED)
    add_link_options("-Wl,--no-undefined")
endif()

```

Reply:
Good point, done.